lib/hooks/queries/useAccountAssetBalances.ts

```diff
@@ -71,36 +72,52 @@ export const useAccountAssetBalances = (
           blockNumber,
         ],
         queryFn: async () => {
-          if (sdk && isRpcSdk(sdk)) {
-            const api = await getApiAtBlock(sdk.api, blockNumber);
-            const balance = !pair.account
-              ? null
-              : await api.query.tokens.accounts(pair.account, pair.assetId);
-
-            return {
-              pair,
-              balance,
-            };
+          const api = await getApiAtBlock(
+            (sdk as RpcContext)!.api,
+            blockNumber,
+          );
+          let balance;
+          if (pair.account) {
+            balance = await api.query.tokens.accounts(
+              pair.account,
+              pair.assetId,
+            );
           }
+
+          return {
+            pair,
+            balance,
+          };
         },
         enabled:
           Boolean(sdk) &&
           isRpcSdk(sdk) &&
           (typeof opts?.enabled === "undefined" ? true : opts?.enabled),
         keepPreviousData: true,
+        initialData: { pair, balance: undefined },
       };
     }),
   });
 
-  const get = (account: string, assetId: AssetId) => {
-    const query = queries.find(
-      (q) =>
-        q.data &&
-        q.data.pair.account === account &&
-        JSON.stringify(q.data.pair.assetId) === JSON.stringify(assetId),
-    );
-    return query;
-  };
+  const get = useMemo(
+    () => (account: string, assetId: AssetId) => {
+      const query = queries.find(
+        (q) =>
+          q.data != null &&
+          q.data.pair.account === account &&
+          JSON.stringify(q.data.pair.assetId) === JSON.stringify(assetId),
+      );
+      if (query?.data === undefined) {
+        throw new Error(
```

Comment:
Are you sure we should throw and error here? Won't that stop code execution?

Reply:
This should never happen in reality though since we're iterating over given `AccountAssetIdPair[]`. If it does happen `get` is used where it should not have been used.